### PR TITLE
Pure JS: Pass real service_url to call credentials

### PR DESCRIPTION
**Pull request by murgatroid99**
I also added the `host` call option, because that impacts the same code paths.

### CODE REVIEW

File: packages/grpc-js-core/src/channel.ts [hunk 1]
```diff
@@ -267,11 +270,12 @@ export class Http2Channel extends EventEmitter implements Channel {
     let finalOptions: CallStreamOptions = {
       deadline: options.deadline === undefined ? Infinity : options.deadline,
       credentials: options.credentials || CallCredentials.createEmpty(),
-      flags: options.flags || 0
+      flags: options.flags || 0,
+      host: options.host || defaultAuthority
```
Comment (kjin) [inline packages/grpc-js-core/src/channel.ts]: Does this need to be `this.defaultAuthority`?

File: packages/grpc-js-core/src/call-credentials-filter.ts [hunk 2]
```diff
@@ -7,13 +7,27 @@ import {BaseFilter, Filter, FilterFactory} from './filter';
 import {Metadata} from './metadata';
 
 export class CallCredentialsFilter extends BaseFilter implements Filter {
-  constructor(private readonly credentials: CallCredentials) {
+  private serviceUrl: string;
+  constructor(
```
Comment (kjin) [inline packages/grpc-js-core/src/call-credentials-filter.ts]: Does `CallCredentials` not contain this information? If it does maybe we should prefer using that.

In any case, the call site for this constructor probably needs to be updated.

File: packages/grpc-js-core/src/call-credentials-filter.ts [hunk 3]
```diff
@@ -7,13 +7,27 @@ import {BaseFilter, Filter, FilterFactory} from './filter';
...
+      private readonly channel: Http2Channel,
+      private readonly host: string,
+      private readonly path: string) {
     super();
+    let splitPath: string[] = path.split('/');
+    let serviceName: string = '';
+    /* The standard path format is "/{serviceName}/{methodName}", so if we split
+     * by '/', the first item should be empty and the second should be the
+     * service name */
+    if (splitPath.length >= 2) {
```
Comment (kjin) [inline packages/grpc-js-core/src/call-credentials-filter.ts]: Is `serviceName` meant to be empty if this is not the case?

Comment (murgatroid99) [reply packages/grpc-js-core/src/call-credentials-filter.ts]: yes

File: packages/grpc-js-core/src/call-credentials-filter.ts (hunk 2 above)
Comment (murgatroid99) [reply packages/grpc-js-core/src/call-credentials-filter.ts]: No, the service URL is a property of the call, not the credentials.

And I fixed the constructor and its call site.

**Pull request merged**